x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTestHelper.java

```diff
@@ -421,10 +476,18 @@ public Authentication build(boolean maybeRunAsIfNotAlready) {
                         // User associated to API key authentication has empty roles
                         user = stripRoles(user);
                         prepareApiKeyMetadata();
-                        authentication = Authentication.newApiKeyAuthentication(
+                        final Authentication apiKeyAuthentication = Authentication.newApiKeyAuthentication(
                             AuthenticationResult.success(user, metadata),
                             ESTestCase.randomAlphaOfLengthBetween(3, 8)
                         );
+                        // Remote access is authenticated via API key, but the underlying authentication instance has a different structure,
+                        // and a different subject type. If remoteAccessAuthentication is set, we transform the API key authentication
+                        // instance into a remote access authentication instance.
+                        // This means the builder only produces remote access authentication instances if `remoteAccess()` was called.
+                        // TODO generalize the builder to randomly build remote access authentication instance, when viable
```

Comment:
I thought I'd be able to generalize the builder in this PR, but this is more involved than I thought -- in a lot of places, it's actually not expected to have a remote_access authentication subject (e.g., anything unrelated to CCS for now); having `AuthenticationTestHelper.builder().build()` produce remote_access authentication instances would require updating quite a lot of tests and make the current PR less focused (also delaying it further). As such, I want to revisit this in a separate PR. Let me know if you think it's better to do this in one go instead.

Reply:
Nice, I'll take a look at `runAs` again and see if that works more cleanly. Otherwise will stick to the current approach. It's a bit clunky but works; since it's test code it'll be esp. easy to iterate on it.